ideascube/serveradmin/catalog.py

```diff
@@ -351,18 +351,31 @@ def _get_package(self, id, source):
         except KeyError:
             raise InvalidPackageType(type)
 
-    def _get_packages(self, id_patterns, source, fail_if_no_match=True):
+    def _get_packages(
+        self, id_patterns, source,
+        fail_if_no_match=True, fail_if_invalid_package=True):
```

Comment:
For the record, though: at the time I introduced this boolean I was still growing the features organically. (I was writing this package manager, after all)

In hindsight, I'm starting to think that validation of the packages should not be done when listing/installing/... as we do now, but instead when refreshing the cache.

That is, when downloading the remote's `yml` file, instead of just storing it as it is, this is when we could try and parse it, look at all the packages, drop the invalid/unsupported ones (warning about them), and then locally keep only the "good" packages.

I guess I just should open a ticket about that, though. :smile: